### PR TITLE
bug: :zap: fix refresh issue in Reports

### PR DESCRIPTION
**Pull request by likhith-deriv**
On refresh of Reports page, the contract id ref is lost which doesn't render Open positions

> Checklist:

- [ ] Ensure the Open positions are displayed on Refresh
- [ ] Ensure the Contract expiry time matches Deriv Air & Deriv App
- [ ] Ensure the Contract position doesn't display **NaN** value
- [ ] On expiry the contract must be removed from the list

### CODE REVIEW

File: src/components/open-position.jsx [hunk 1]
```diff
@@ -1,64 +1,78 @@
...
+import { For, Show, onCleanup, onMount } from "solid-js";
+import { forgetAll, wait } from "../utils/socket-base";
 import {
   open_contract_ids,
-  setOpenContractId,
   open_contract_info,
+  setOpenContractId,
   setOpenContractInfo,
 } from "../stores";
+
```
Comment (farrah-deriv) [inline src/components/open-position.jsx]: please remove this

File: src/components/open-position.jsx [hunk 2]
```diff
@@ -1,64 +1,78 @@
...
+  if (!open_contract_ids().includes(contract_info.contract_id)) {
+    setOpenContractId([...open_contract_ids(), contract_info.contract_id]);
+  }
+  if (contract_info.is_expired) {
+    const new_set = open_contract_ids().filter(
+      (ctd_id) => contract_info.contract_id !== ctd_id
+    );
+    setOpenContractId(new_set);
+    delete open_contract_info()[contract_info.contract_id];
+    setOpenContractInfo({ ...open_contract_info() });
```
Comment (farrah-deriv) [inline src/components/open-position.jsx:40]: since we are not updating anything here, do we need to call this?

File: src/utils/socket-base.js [hunk 3]
```diff
@@ -1,5 +1,7 @@
 import { getAppId, getSocketUrl } from "Utils/config.js";
+
```
Comment (farrah-deriv) [inline src/utils/socket-base.js:2]: please remove this

**Review by farrah-deriv — COMMENTED** [review]
Expired contracts are showing under Open positions tab on the testlink. Please check.

File: src/components/open-position.jsx (hunk 1 above)
Comment (likhith-deriv) [reply src/components/open-position.jsx]: It is getting applied on save :)

File: src/utils/socket-base.js (hunk 3 above)
Comment (likhith-deriv) [reply src/utils/socket-base.js:2]: It is getting applied on save :)

**Review by farrah-deriv — APPROVED** [review]

**Review by vinu-deriv — APPROVED** [review]

**Pull request merged**